VoodooRMI/Functions/Input/RMIGPIOFunction.cpp

```diff
@@ -0,0 +1,159 @@
+/* SPDX-License-Identifier: GPL-2.0-only
+ * Copyright (c) 2021 Avery Black
+ * Ported to macOS from linux kernel, original source at
+ * https://github.com/torvalds/linux/blob/master/drivers/input/rmi4/F30.c
+ *
+ * Copyright (c) 2012-2016 Synaptics Incorporated
+ */
+
+#include "RMIGPIOFunction.hpp"
+
+OSDefineMetaClassAndStructors(RMIGPIOFunction, RMIFunction)
+#define super RMIFunction
+
+bool RMIGPIOFunction::attach(IOService *provider)
+{
+    int error;
+
+    error = initialize();
+    if (error)
+        return false;
+
+    super::attach(provider);
+
+    return true;
+}
+
+int RMIGPIOFunction::config()
```

Comment:
It seems that `F3A` doesn't do this step?

Reply:
No, probably not needed in F03 either. Mostly if we decide to do any configuration like for LEDs. That's mostly an F03 thing though.